src/network_models/power_flow_evaluation.jl

```diff
@@ -42,20 +46,44 @@ pf_input_keys(::PFS.ACPowerFlowData) =
     [:active_power, :reactive_power, :voltage_angle_opf, :voltage_magnitude_opf]
 pf_input_keys(::PFS.PSSEExporter) =
     [:active_power, :reactive_power, :voltage_angle_export, :voltage_magnitude_export]
+pf_input_keys_hvdc_pst(::PFS.PowerFlowData) = DataType[]
+pf_input_keys_hvdc_pst(::PFS.ACPowerFlowData) = [
+    :active_power_hvdc_from_to,
+    :active_power_hvdc_to_from,
+    :active_power_pst_from_to,
+    :active_power_pst_to_from,
+]
 
-# Maps the StaticInjection component type by name to the
-# index in the PowerFlow data arrays going from Bus number to bus index
-function _make_temp_component_map(pf_data::PFS.PowerFlowData, sys::PSY.System)
+# Generalized function to create component maps by name to the index in the PowerFlowData bus arrays
+function _make_temp_component_map(
+    pf_data::PFS.PowerFlowData,
+    sys::PSY.System,
+    component_type::DataType,
+    get_bus_func::Function,
+)
     temp_component_map = Dict{DataType, Dict{String, Int}}()
-    available_injectors = PSY.get_available_components(PSY.StaticInjection, sys)
+    components = PSY.get_available_components(component_type, sys)
     bus_lookup = PFS.get_bus_lookup(pf_data)
-    for comp in available_injectors
+    for comp in components
         comp_type = typeof(comp)
         bus_dict = get!(temp_component_map, comp_type, Dict{String, Int}())
-        bus_number = PSY.get_number(PSY.get_bus(comp))
-        bus_dict[get_name(comp)] = bus_lookup[bus_number]
+        bus_number = PSY.get_number(get_bus_func(comp))
```

Comment:
What's this `get_bus_func`? I don't like to use anonymous functions in hot loops because they can't get pre-compiled.